product_cost_rollup_to_bom/models/product.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (C) 2021, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+from datetime import datetime
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    std_cost_update_date = fields.Datetime(
+        string="Standard Cost Update Date",
+        copy=False,
+        help="Last time the standard cost was performed on this product.",
+    )
+
+    def action_bom_cost(self):
+        real_time_products = self.filtered(
+            lambda p: p.valuation == "real_time" and p.valuation == "fifo"
+        )
+        if real_time_products:
+            raise UserError(
+                _(
+                    "The costing method on some products %s is FIFO."
+                    " The cost will be computed during manufacturing process."
+                    " Use Standard Costing to update BOM cost manually."
+                )
+                % (real_time_products.mapped("display_name"))
+            )
+        # else:
+        boms_to_recompute = self.env["mrp.bom"].search(
+            [
+                "|",
+                ("product_id", "in", self.ids),
+                "&",
+                ("product_id", "=", False),
+                ("product_tmpl_id", "in", self.mapped("product_tmpl_id").ids),
+            ]
+        )
+        for product in self:
+            new_price = product._set_price_from_bom(boms_to_recompute) or 0.0
+            # FIXME: precision rounding should be taken from configs
+            if product.cost_method == "standard" and not float_is_zero(
+                new_price - product.standard_price, precision_rounding=2
+            ):
+                product._change_standard_price(new_price)
+                product.std_cost_update_date = datetime.now()
+                if product.product_tmpl_id.product_variant_count == 1:
+                    _logger.info(
+                        "Product : %s Standard Price: %s ",
+                        product.default_code,
+                        str(product.product_tmpl_id.standard_price),
+                    )
+                else:
+                    _logger.info(
+                        "Product : %s Standard Price: %s ",
+                        product.default_code,
+                        str(product.standard_price),
+                    )
+
+    def _set_price_from_bom(self, boms_to_recompute=False):
+        self.ensure_one()
+        bom = self.env["mrp.bom"]._bom_find(product=self)
+        if bom:
+            self.standard_price = self.with_context(cost_all=True)._compute_bom_price(
+                bom, boms_to_recompute=boms_to_recompute
+            )
+            bom.std_cost_update_date = datetime.now()
+
+    def _compute_bom_price(self, bom, boms_to_recompute=False):
```

Comment:
@Chandresh-OSI Let's start my having an email template instead of an email config, and have the Update From BoM button available also for Standard Cost products. Then we'll see f any of the compute price code s really needed.